### PR TITLE
Support ASP.NET Core 3.0

### PR DESCRIPTION
**Pull request by martincostello**
* Update to target ASP.NET Core 3.0.
  * Deprecate the `HttpClient` property on the options class and redirect it to the base class' `Backchannel` property which serves the same purpose. [See here](https://github.com/aspnet/AspNetCore/blob/62351067ff4c1401556725b401478e648b66acdc/src/Security/Authentication/OAuth/src/OAuthPostConfigureOptions.cs#L29-L35) for the OAuth equivalent.

### CODE REVIEW

**Comment by martincostello** [comment]
Rebase onto #63 if/when merged.

**Comment by martincostello** [comment]
~Need to rebase onto `dev` now that #63 has been merged.~

File: NuGet.config [hunk 1]
```diff
@@ -1,9 +1,27 @@
 <?xml version="1.0" encoding="utf-8"?>
 <configuration>
   <packageSources>
+    <clear />
```
Comment (martincostello) [inline NuGet.config]: Will revert this once the final packages get to NuGet.org.

File: test/AspNet.Security.OpenId.Providers.Tests/Infrastructure/ApplicationFactory.cs [hunk 2]
```diff
@@ -85,26 +85,31 @@ private static void ConfigureApplication(IApplicationBuilder app)
...
-                    }
-                    else
-                    {
-                        await context.ChallengeAsync();
-                    }
-                }));
+            app.UseRouting();
+
+            app.UseAuthentication()
+               .UseEndpoints(endpoints =>
```
Comment (davidfowl) [inline test/AspNet.Security.OpenId.Providers.Tests/Infrastructure/ApplicationFactory.cs:94]: Add the UseAuthorization middleware as well?

Comment (martincostello) [reply test/AspNet.Security.OpenId.Providers.Tests/Infrastructure/ApplicationFactory.cs:94]: Ah yes, missed that from the tests. Thanks!

**Pull request merged**